src/core/util/options.js

```diff
@@ -338,6 +338,16 @@ function normalizeInject (options: Object, vm: ?Component) {
   const normalized = options.inject = {}
   if (Array.isArray(inject)) {
     for (let i = 0; i < inject.length; i++) {
+      if (typeof inject[i] !== 'string') {
+        if (process.env.NODE_ENV !== 'production') {
+          warn(
+            `Invalid value for option "inject": expected an Array of string, ` +
+              `but found ${toRawType(inject[i])}.`,
+            vm
+          );
+        }
+        continue;
+      }
```

Comment:
Let's refactor this like the following snippet (cache `inject[i]` ahead and reverse the if logic): 

https://github.com/vuejs/vue/blob/4d6e438d937dbe8f2a2fa0650855ff81ed9f62b9/src/core/util/options.js#L306-L312

So that we can get rid of the `continue` here.